frontend/packages/console-app/src/components/modals/clone/clone-pvc-modal.tsx

```diff
@@ -70,74 +84,87 @@ const ClonePVCModal = withHandlePromise((props: ClonePVCModalProps) => {
+      (cloneResource) => {
+        close();
+        history.push(resourceObjPath(cloneResource, referenceFor(cloneResource)));
+      },
+    );
   };
 
   return (
     <Form onSubmit={submit}>
       <div className="modal-content modal-content--no-inner-scroll">
         <ModalTitle>Clone</ModalTitle>
         <ModalBody>
-          <FormGroup label="Name" isRequired fieldId="ceph-clone-pvc-modal__name">
+          <FormGroup label="Name" isRequired fieldId="clone-pvc-modal__name">
             <TextInput
-              isRequired
               type="text"
-              name="ceph-clone-pvc-modal__name"
+              className="clone-pvc-modal__name--margin"
               value={clonePVCName}
               onChange={setClonePVCName}
+              aria-label="Clone Pvc"
+            />
+          </FormGroup>
+          <FormGroup label="Size" isRequired fieldId="clone-pvc-modal__size">
+            <RequestSizeInput
+              name="requestSize"
+              onChange={requestedSizeInputChange}
+              defaultRequestSizeUnit={requestedUnit}
+              defaultRequestSizeValue={requestedSize}
+              dropdownUnits={dropdownUnits}
+              required
             />
           </FormGroup>
-          <div className="ceph-clone-pvc-modal__details">
+          <div className="co-clone-pvc-modal__details">
             <p className="text-muted">PVC Details</p>
-            <div className="ceph-clone-pvc-modal__details-section">
+            <div className="clone-pvc-modal__details-section">
               <div>
                 <div>
-                  <p className="ceph-clone-pvc-modal__pvc-details">Namespace</p>
+                  <p className="clone-pvc-modal__pvc-details">Namespace</p>
                   <p>
                     <ResourceIcon kind="Namespace" />
                     {resource.metadata.namespace}
                   </p>
                 </div>
                 <div>
-                  <p className="ceph-clone-pvc-modal__pvc-details">Storage Class</p>
+                  <p className="clone-pvc-modal__pvc-details">Storage Class</p>
                   <p>
-                    {resource.spec.storageClassName ? (
-                      <>
-                        <ResourceIcon kind="StorageClass" />
-                        {resource.spec.storageClassName}
-                      </>
-                    ) : (
-                      'None'
-                    )}
+                    <ResourceIcon kind="StorageClass" />
+                    {resource.spec?.storageClassName || '-'}
                   </p>
                 </div>
               </div>
               <div>
                 <div>
-                  <p className="ceph-clone-pvc-modal__pvc-details">Requested Capacity</p>
-                  <p>{resource.spec.resources.requests.storage}</p>
+                  <p className="clone-pvc-modal__pvc-details">Requested Capacity</p>
+                  <p>{getRequestedPVCSize(resource)}</p>
```

Comment:
nit: to maintain the console standard, let's convert the size to G/T/MiB before displaying.